File: backend/src/main/java/pika/hungt1/dx/entity/SalesOrderItem.java
package pika.hungt1.dx.entity;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.fasterxml.jackson.annotation.JsonBackReference;

@Entity
@Table(name = "sales_order_items")
@Data @NoArgsConstructor @AllArgsConstructor
public class SalesOrderItem {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name="order_id")
    private SalesOrder order;


    @ManyToOne
    @JoinColumn(name="product_id")
    private Product product;

    @Column(nullable=false)
    private Integer quantity;

    @Column(name="unit_price", nullable=false)
    private BigDecimal unitPrice;
}

File: backend/Models/SalesOrderItem.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models
{
    [Table("sales_order_items")]
    public class SalesOrderItem
    {
        [Key] public int Id { get; set; }

        public int OrderId { get; set; }

        [ForeignKey("OrderId")]
        public SalesOrder Order { get; set; }

        public int ProductId { get; set; }

        [ForeignKey("ProductId")]
        public Product Product { get; set; }

        [Required]
        public int Quantity { get; set; }

        [Required]
        public decimal UnitPrice { get; set; }
    }
}
